src/textual/widgets/_collapsible.py

```diff
@@ -1,87 +1,93 @@
 from __future__ import annotations
 
-from textual.widget import Widget
+from rich.console import RenderableType
+from rich.text import Text
 
 from .. import events
 from ..app import ComposeResult
-from ..containers import Container, Horizontal
+from ..binding import Binding
+from ..containers import Container
+from ..css.query import NoMatches
 from ..message import Message
 from ..reactive import reactive
 from ..widget import Widget
-from ..widgets import Label
 
-__all__ = ["Collapsible"]
+__all__ = ["Collapsible", "CollapsibleTitle"]
 
 
-class Collapsible(Widget):
-    """A collapsible container."""
-
-    collapsed = reactive(True)
+class CollapsibleTitle(Widget, can_focus=True):
+    """Title and symbol for the Collapsible."""
 
     DEFAULT_CSS = """
-    Collapsible {
-        width: 1fr;
+    CollapsibleTitle {
+        width: auto;
         height: auto;
+        padding: 0 1 0 1;
+    }
+
+    CollapsibleTitle:hover {
+        background: $foreground 10%;
+        color: $text;
+    }
+
+    CollapsibleTitle:focus {
+        background: $accent;
     }
     """
 
-    class Title(Horizontal):
-        DEFAULT_CSS = """
-        Title {
-            width: 100%;
-            height: auto;
-        }
+    BINDINGS = [Binding("enter", "toggle", "Toggle collapsible", show=False)]
 
-        Title:hover {
-            background: grey;
-        }
+    collapsed = reactive(False)
 
-        Title .label {
-            padding: 0 0 0 1;
-        }
+    def __init__(
+        self,
+        *,
+        label: str,
+        collapsed_symbol: str,
+        expanded_symbol: str,
+        collapsed: bool,
+    ) -> None:
+        super().__init__()
+        self.collapsed_symbol = collapsed_symbol
+        self.expanded_symbol = expanded_symbol
+        self.label = label
+        self.collapse = collapsed
 
-        Title #collapsed-symbol {
-            display:none;
-        }
+    class Toggle(Message):
+        """Request toggle."""
 
-        Title.-collapsed #expanded-symbol {
-            display:none;
-        }
+    async def _on_click(self, event: events.Click) -> None:
+        """Inform ancestor we want to toggle."""
+        event.stop()
+        self.post_message(self.Toggle())
 
-        Title.-collapsed #collapsed-symbol {
-            display:block;
-        }
-        """
+    def action_toggle(self) -> None:
+        """Toggle the state of the parent collapsible."""
+        self.post_message(self.Toggle())
+
+    def render(self) -> RenderableType:
+        """Compose right/down arrow and label."""
+        if self.collapsed:
+            return Text.assemble(self.collapsed_symbol, " ", self.label)
+        else:
+            return Text.assemble(self.expanded_symbol, " ", self.label)
```

Comment:
Doesn't this get rerun every single time we need to show the title, even if the collapsed state didn't change?
Couldn't we cache this if we inherited from `Static`?

Reply:
Only when it is repainted. It's such a trivial function that caching the renderable may not even be a win.

But I have noticed, that it is doing a little more work that required!